src/test/java/com/microsoft/azure/relay/SendReceiveTest.java

```diff
@@ -170,6 +175,34 @@ public void websocketMultipleSenderTest() throws URISyntaxException {
 		assertEquals("Sendner did not receive expected number of messages.", numberOfSenders, senderReceiveCount);
 	}
 	
+	@Test
+	public void websocketWriteReceiveAllByteValuesTest() throws URISyntaxException {
+		CompletableFuture<Void> listenerTask = listener.acceptConnectionAsync().thenCompose(connection -> {
+			return connection.readAsync()
+				.thenCompose(bufferReceived -> {
+					byte[] byteSet = getByteSet();
+					assertTrue("Websocket listener did not receive the expected bytes.", Arrays.equals(byteSet, bufferReceived.array()));
+					return connection.writeAsync(ByteBuffer.wrap(byteSet));
+				})
+				.thenCompose($void -> {
+					return connection.closeAsync(new CloseReason(CloseCodes.NORMAL_CLOSURE, "Normal closure from listener"));
+				});
+		});
+		
+		HybridConnectionClient newClient = new HybridConnectionClient(new URI(TestUtil.RELAY_NAMESPACE_URI + TestUtil.ENTITY_PATH), tokenProvider);
```

Comment:
>newClient [](start = 25, length = 9)

should this test ideally close the HCClient no matter what?

Reply:
HCClient has no close(), closing the connection it creates instead.

---
In reply to: [294380700](https://github.com/Azure/azure-relay-java/pull/47#discussion_r294380700) [](ancestors = 294380700)